api/envoy/config/route/v3/route_components.proto

```diff
@@ -2376,7 +2376,6 @@ message InternalRedirectPolicy {
 // :ref:`Route.typed_per_filter_config<envoy_v3_api_field_config.route.v3.Route.typed_per_filter_config>`,
 // or :ref:`WeightedCluster.ClusterWeight.typed_per_filter_config<envoy_v3_api_field_config.route.v3.WeightedCluster.ClusterWeight.typed_per_filter_config>`
 // to add additional flags to the filter.
```

Comment:
I think it just add some more non-filter-specific flags/control options to custom the filter's behavior in specific route.

Reply:
> It could be either way. It really depends on how the filter is implemented. For some filters, the semantic is that the per-route/vh config completely overrides the top-level config; for other filters, the two configs are merged together in some appropriate way.

**From perspective of external filter manager, the route level per filter config just a supplement to the filter config in the HCM**. But in some filters' implementations, the `supplement` may be used by priority.